crates/ibc/src/core/ics04_channel/handler/acknowledgement.rs

```diff
@@ -105,9 +106,11 @@ pub fn process<Ctx: ChannelReader>(
 
     output.log("success: packet ack");
 
-    output.emit(IbcEvent::AcknowledgePacket(AcknowledgePacket {
-        packet: packet.clone(),
-    }));
+    output.emit(IbcEvent::AcknowledgePacket(AcknowledgePacket::new(
```

Comment:
Unrelated to this PR, but I think we need to check for redundant relays and treat them as noops (although an event must be emitted) -> https://github.com/cosmos/ibc-go/blob/a0e59b8e7a2e1305b7b168962e20516ca8c98fad/modules/core/04-channel/keeper/packet.go#L419-L426

Reply:
created #213